app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/payloadValidationActions.scala

```diff
@@ -84,39 +84,35 @@ abstract class PayloadValidationAction(val xmlValidationService: XmlValidationSe
 
   override def refine[A](ar: AuthorisedRequest[A]): Future[Either[Result, ValidatedPayloadRequest[A]]] = {
     implicit val implicitAr = ar
+    lazy val errorMessage = "Request body does not contain a well-formed XML document."
 
-    validateXml
-  }
+    ar.body match {
+      case content: AnyContent =>
+        content.asXml
+          .map(validateXml(_))
+          .getOrElse(Future.successful(Left(errorResponse(errorMessage))))
 
-  private def validateXml[A](implicit ar: AuthorisedRequest[A]): Future[Either[Result, ValidatedPayloadRequest[A]]] = {
-    lazy val errorMessage = "Request body does not contain a well-formed XML document."
-    lazy val errorNotWellFormed = ErrorResponse.errorBadRequest(errorMessage).XmlResult.withConversationId
+      case _ => Future.successful(Left(errorResponse(errorMessage)))
+    }
+  }
 
-    def validate(xml: NodeSeq): Future[Either[Result, ValidatedPayloadRequest[A]]] =
-      xmlValidationService.validate(xml).map{ _ =>
+  private def validateXml[A](xml: NodeSeq)(implicit ar: AuthorisedRequest[A]): Future[Either[Result, ValidatedPayloadRequest[A]]] = {
+    xmlValidationService.validate(xml)
+      .map { _ =>
         logger.debug("XML payload validated")
         Right(ar.toValidatedPayloadRequest(xml))
       }
       .recover {
         case saxe: SAXException =>
-          val msg = "Payload did not pass validation against the schema."
+          val msg = "Payload is not valid according to schema"
```

Comment:
I don't usually change the text of an error msg. Someone might be using the existing text.